local/local.go

```diff
@@ -71,28 +76,107 @@ func FromBaseImage(imageName string) ImageOption {
 	}
 }
 
-func NewImage(repoName string, dockerClient client.CommonAPIClient, ops ...ImageOption) (*Image, error) {
+func WithPlatform(platform imgutil.Platform) InitialImageOption {
+	return func(i *Image) (*Image, error) {
+		if platform.OS != "" && platform.OS != i.inspect.Os {
+			return nil, fmt.Errorf(`invalid os: platform os "%s" must match the daemon os "%s"`, platform.OS, i.inspect.Os)
+		}
+
+		i.inspect.Architecture = platform.Architecture
+		i.inspect.OsVersion = platform.OSVersion
+
+		i.platform = platform
+
+		return i, nil
+	}
+}
+
+func NewImage(repoName string, dockerClient client.CommonAPIClient, ops ...interface{}) (*Image, error) {
```

Comment:
This is getting complicated enough that we should definately add a doc string describing the behavior of this function, regardless of whether we make the change suggested above.

Reply:
Potentially better idea to replace the interfaces!

Let's copy GGCR and make our `ImageOption` mutate an internal options struct, instead of directly mutating the image. We can set options fields like `baseImage` and `platform` and then do sane things with the results once all the options are applied.

https://github.com/google/go-containerregistry/blob/main/pkg/v1/remote/options.go#L28-L40

This also solves the problem with the platform field described in a different comment b/c we wouldn't need to save the platform as separate field on the Image.

Reply:
Sounds good! I'll change the `ImageOption`s to not modify the image, so I can make them all the same type again.

And I'll add the missing doc strings for them, including the caveats from this PR thread - totally missed that one